airflow/config_templates/default_airflow.cfg

```diff
@@ -524,9 +524,6 @@ username =
 password =
 
 [operators]
-# The default owner assigned to each new operator, unless
-# provided explicitly or passed via ``default_args``
-default_owner = airflow
```

Comment:
What will happens to users who customized their airflow.cfg with 
`default_owner = something`

This will no longer work for this... so isn't this a breaking change that we can do only in Airflow 3?

Reply:
Yes that's correct.

Reply:
Why are we deprecating this at all? I actually like the proposal of not using "airflow" as the default, but I don't see why that means we have to also deprecate the ability for folks to set a default owner at the config level at all? Perhaps they have a default that makes sense for their workflow or organizational structure.

IMHO, this config option should just no longer be a "mandatory" value, but still available to use.

Reply:
I did some research within Astronomer and there's not a single customer setting this option. I figured we might as well remove it for the sake of simplifying Airflow configuration. WDYT?

Either way (remove/change to `None`), this should be treated as a breaking change.

Reply:
Following the discussion in https://github.com/apache/airflow/pull/27067#discussion_r1020614511 - this seems like a good candidate to remove IF we agree that we can treat "breaking" in a less strict and more "how likely it's going to break other's workflow" way. 

Very good example of a case where we could take a risk and classify it as "non-breaking" (even if it is technically a removal).

Reply:
Got a bit lost in that very long message. Are you proposing to discuss the definition of "breaking change"?

In this PR it's clear --> user facing feature, so treat as breaking change, so change in Airflow 3.0.

Reply:
I agree with Jarek that we could maybe convince ourselves that this is not a hard breaking change, but I actually think this feature is somewhat useful and don't see the motivation for getting rid of it.

We can still keep the ability to set that a globally default user, but just make the config no longer mandatory (since it is currently mandatory). That way you can remove it from the newly published configs, but those users who still have this setting in their configs will not have any regressions and that feature will continue to work.